(DOCSP-34624): Device SDK Landing Page and Intro updates (#3092)
mongodb/docs-realm: source/index.txt

:hidefeedback: header
:noprevnext:

==============================

> Welcome to the Device SDK Docs

Reviewer: should this be "Atlas Device SDK Docs"?

Reviewer 2: Not having "Atlas Device SDK" in top-level titles and headings feels odd. If that's the product name and what people will see in Atlas, they could get confused by seeing the more generic "Device SDK".

==============================

.. meta:: 

>    :description: Device SDK is an offline-first, cloud-syncable database for mobile, web, desktop, and IoT apps.

Reviewer: IDK if we're following it exactly, but per that renaming doc:
[after the first formal reference] feel free to refer to them as “the SDKs” or “the SDK” 
```suggestion
   :description: The SDKs are an offline-first, cloud-syncable database for mobile, web, desktop, and IoT apps.
```

Reviewer 2: I think the "full name, then 'SDK' or 'SDKs'" construction could work well. The repetition of "Device SDK" everywhere feels odd. (though at least it improves the realm/Realm thing a bit. Maybe.)

Shortening subsequent references to just "SDK" would be wonderful.

Reviewer: might be just me, but the singular "Device SDK is ... " without a definite article also feels odd

Reviewer 3: Probably best to include the full name (Atlas Device SDK) for the meta description.

Author: I'm adjusting the naming per our discussion yesterday. But FWIW, the meta:description should always include the full name, because that's what shows up in Google search results. We shouldn't use the short version there. It doesn't actually show up on the page.

We can use the SDK short name after the first "visible" full product name on the page (after the title). I'll share the formalized doc from our convo yesterday after I apply the changes to this PR and make sure I'm clear on everything.

.. introduction::

   Atlas Device SDK is a suite of app development tools optimized for 
   mobile and edge device persistence and data access. Use the SDKs to 

>    build data-driven mobile, edge, web, desktop, and IoT apps.

Author: Optimizing for devices (mobile, specifically) was the driving force behind the Realm product development. A lot of the criteria that went into developing Realm was specific to mobile development - i.e. fast, lightweight, embedded, object-based, supports low power consumption, etc. Persistence is the purpose that it serves, but the optimization was for device.

Do you think I need to be more verbose here to clarify that detail?

   .. button:: Get started with a tutorial
      :uri: https://mongodb.com/docs/atlas/app-services/get-started/

   :doc:`Read the 5-Minute Introduction <introduction>`

.. image:: /images/hero.png
   :alt: Atlas Device SDK landing page hero image
   :class: hero-img

> The SDKs provide tools to read and write Atlas data from device. Your app 

Author: I am happy to incorporate these suggestions, but I wonder if we need to align as a team around the use of singular/plural device/devices and whether we need an article before "device." My choices here weren't overlooked grammar issues - they were intentional. I think these changes could get pretty inconsistent if we're not all on the same page as we incorporate the new naming.

can sync automatically with MongoDB Atlas and other devices using Device Sync. 
You can access the MongoDB GraphQL API, or call Atlas Functions from a 
device. The device persistence layer is Realm, an embedded, object-oriented 
database that lets you build real-time, offline-first applications. 

> The SDKs support most popular languages, frameworks, and platforms. Each 

Author: I see what you're going for, but I don't love the passive "There are..."

What do you think about:

> The SDK provides packages for most popular...

Or even just:

> We provide SDKs for most popular...

.. introduction::

>    Device SDK is a suite of app development tools whose core is a 
>    mobile-first database designed for modern, data-driven applications. 
>    Use Device SDK to build mobile, web, desktop, and IoT apps.

Reviewer 2: This first sentence is doing a lot of work. Can we simplify?

```suggestion
   Device SDK is a suite of app development tools with a core
   mobile-first database. Use Device SDK to build modern
   and data-driven mobile, web, desktop, and IoT apps.
```

Reviewer 2: Alternatively, viewing this from an Atlas-first perspective, maybe something like this is better? Also, you can't really build an app with the SDK. Rather, you integrate it with an app so that you can work with data that lives in Atlas, right?

```suggestion
   Device SDK is a suite of app development tools that
   translate Atlas data to an on-device, mobile-first database. 
   Use Device SDK to integrate Atlas data with, web, desktop,
   and IoT apps.
```

Author: Went with a compromise between these two things and something slightly different based on our convo yesterday. Curious for your thoughts!

   :alt: Realm landing page hero image
   :class: hero-img

> Device SDK provides tools to read and write data from device, sync with 
> MongoDB Atlas, access the MongoDB GraphQL API, or call Atlas Functions from
> a device. Its local persistence layer is Realm, an embedded, object-oriented 
> database that lets you build real-time, always-on applications. 

Reviewer 2: An Atlas-first suggestion:

```suggestion
Device SDK provides tools to access and persist Atlas data on many kinds
of devices. The device persistence layer is called Realm, an embedded,
object-oriented database for real-time always-on applications. Realm works
with Device Sync to keep all devices updated.

You can also access your Atlas data from a device in other ways, like 
using the MongoDB GraphQL API or Atlas Functions.
```

Author: I tweaked this a little based on our convo yesterday, and more thinking about it this AM. What do you think of the changes I made?
